libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/MapboxRerouteController.kt

```diff
@@ -42,6 +45,8 @@ internal class MapboxRerouteController @VisibleForTesting constructor(
 
     private var requestId: Long? = null
 
+    private var rerouteScope = threadController.getMainScopeAndRootJob().scope
```

Comment:
Instead of creating a new scope, you can save a reference to the reroute job:
```kotlin
private var rerouteJob: Job? = null
```
```kotlin
rerouteJob?.cancel()
rerouteJob = null
```
```kotlin
rerouteJob = mainJobController.scope.launch {
    // perform reroute
}
```
I think it is a more efficient approach, but I don't have anything to back it up.